src/xtgeo/xyz/points.py

```diff
@@ -1,52 +1,227 @@
+            zonelist=zonelist,
+            use_undef=use_undef,
+        )
+
+        if wp is not None:
+            dflist.append(wp)
+
+    dfr = pd.concat(dflist, ignore_index=True)
+
+    attrs = {}
+    for col in dfr.columns:
+        if col == "Zone":
+            attrs[col] = "int"
+        elif col == "ZoneName":
+            attrs[col] = "str"
+        elif col == "WellName":
+            attrs[col] = "str"
+        else:
+            attrs[col] = "float"
+
+    return {"values": dfr, "attributes": attrs}
+
+
+def _wells_dfrac_importer(
+    wells: List[xtgeo.Well],
+    dlogname: str,
+    dcodes: List[int],
+    incl_limit: float = 90,
+    count_limit: int = 3,
+    zonelist: list = None,
+    zonelogname: str = None,
+) -> Dict:
+
+    """General function, get fraction of discrete code(s) e.g. facies per zone."""
+
+    dflist = []
+    for well in wells:
+        wpf = well.get_fraction_per_zone(
+            dlogname,
+            dcodes,
+            zonelist=zonelist,
+            incl_limit=incl_limit,
+            count_limit=count_limit,
+            zonelogname=zonelogname,
+        )
+
+        if wpf is not None:
+            dflist.append(wpf)
+
+    dfr = pd.concat(dflist, ignore_index=True)
+
+    attrs = {}
+    for col in dfr.columns[3:]:
+        if col.lower() == "zone":
+            attrs[col] = "int"
+        elif col.lower() == "zonename":
+            attrs[col] = "str"
+        elif col.lower() == "wellname":
+            attrs[col] = "str"
+        else:
+            attrs[col] = "float"
+
+    return {
+        "values": dfr,
+        "attributes": attrs,
+        "zname": "DFRAC",
+    }
 
 
-def points_from_file(wfile, fformat="xyz"):
+def points_from_file(pfile: Union[str, pathlib.Path], fformat: Optional[str] = "guess"):
     """Make an instance of a Points object directly from file import.
 
+    Supported formats are:
+
+        * 'xyz' or 'poi' or 'pol': Simple XYZ format
+        * 'zmap': ZMAP line format as exported from RMS (e.g. fault lines)
+        * 'rms_attr': RMS points formats with attributes (extra columns)
+        * 'guess': Try to choose file format based on extension
+
+
     Args:
-        mfile (str): Name of file
-        fformat (str): See :meth:`Points.from_file`
+        pfile: Name of file or pathlib object.
+        fformat: File format, xyz/pol/... Default is `guess` where file
+            extension or file signature is parsed to guess the correct format.
 
     Example::
 
         import xtgeo
         mypoints = xtgeo.points_from_file('somefile.xyz')
     """
-    obj = Points()
-
-    obj.from_file(wfile, fformat=fformat)
-
-    return obj
+    return Points(**_file_importer(pfile, fformat=fformat))
 
 
 def points_from_roxar(
-    project, name, category, stype="horizons", realisation=0, attributes=False
+    project,
+    name: str,
+    category: str,
+    stype: str = "horizons",
+    realisation: int = 0,
+    attributes: bool = False,
 ):
```

Comment:
The Typing is used a bit non-consistent here. I used e.g.
`attributes: Optional[bool] = False`
instead of here:
`attributes: bool = False`

The first one will in the auto - documention add the Optional mark which may be preferred. Examples here:
![Screenshot_2021-12-01_11-13-00](https://user-images.githubusercontent.com/16700455/144216543-6d50a940-92dd-40ec-87e6-d23bc64adbc8.png)
![Screenshot_2021-12-01_11-14-25](https://user-images.githubusercontent.com/16700455/144216546-b1ddd791-d7cb-4753-b8c4-d62ad2bb7ac9.png)

Reply:
`Optional[int]` means that the parameter can be `None` or an `int`, which isn't the case here.

Reply:
ok